fixtures/workers-with-assets/tests/index.test.ts

```diff
@@ -266,3 +266,30 @@ describe("[Workers + Assets] dynamic site", () => {
 		`);
 	});
 });
+
+describe("[Workers + Assets] logging", () => {
+	it("should log _headers and _redirects parsing", async ({ expect }) => {
+		const { ip, port, stop, getOutput } = await runWranglerDev(
+			resolve(__dirname, ".."),
+			["--port=0", "--inspector-port=0"]
+		);
+		expect(getOutput()).toContain(
+			`[wrangler:info] ✨ Parsed 2 valid redirect rules.`
+		);
+		expect(getOutput()).toContain(
+			`[wrangler:info] ✨ Parsed 1 valid header rule.`
+		);
+		await stop();
```

Comment:
The `stop` might not be called if the assertion above failed. How about making it `onTestFinished(() => stop())`?

Ref: https://vitest.dev/api/#ontestfinished

Reply:
8cc338c11